### DISCUSSION TITLE
Difference between Normal Scikit Learn and Intel Extension of Scikit Learn

### DISCUSSION THREAD

**Discussion by ameervalki** [intel/scikit-learn-intelex#1687] [Q&A] [answered]
Hi

Can someone please clarify the difference between Parallelism in Stock Scikit Learn and the acceleration provided by Intel extension of Scikit Learn. 

What is the base of Intel extension of Scikit Learn?

How it gives more acceleration than Stock Scikit Learn?

Also how to use specific number of cores in Intel extension of scikit learn if I dont want all of my physical cores to be used

**Comment by icfaust** [comment, marked as answer]
Hello @ameervalki , thank you for the question.  Scikit-learn uses joblib and in some cases a special n_jobs keyword argument to control multithreading in their estimators.  Scikit-learn-intelex makes all estimators available for parallelization by adding an n_jobs keyword to all estimators which is used by some methods.  Additionally, we have optimized C++ routines from the oneDAL project which have been designed for intel hardware.  This is dependent on intel's MKL and TBB products (among others). Scikit-learn-intelex installs ontop of you scikit-learn installation and introduces these alternate routines when possible.  You can use our product either by using the ```patch_sklearn``` routine, or by directly importing our estimators.
